### ISSUE TITLE
Auto-type resets its own macOS permission, than fails

### ISSUE THREAD

**Issue by jrobert-gh** [keepassxreboot/keepassxc#5945]
## Overview
Auto-type requested permission to control my computer. I granted it and tried again. A-T requested permission again. Indeed, that permission had been reset! A-T had been working correctly in v2.3.1. The reported behavior began when I upgraded to v2.6.3. macOS 10.14.6.

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. Open KeePassXC
2. Select an entry that has a username, password, and a URL.
3. Double-click the URL. The browser launches and opens the URL.
4. Click the KeePassXC Auto-type icon. A KeePassXC dialog appears: "KeePassXC requires the Accessibility permission ... ", and a system dialog appears: "KeePassXC would like to control this computer ... ".
5. Open System Preferences | Security and Privacy |  Accessibility. In the right-pane, KeePassXC's permission has been unticked.

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
At step 4 above, I expected 1) Auto-type to run and log my browser into the selected site; and 2) my System Preferences to remain unchanged.

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
[ As described in steps 4 and 5, above ]

## Context
[NOTE]: # ( Give us any additional information you may have. )

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.6.3
Revision:  beae186

[NOTE]: # ( Pick choices based on your environment )
Operating System: macOS
Desktop Env: macOS
Windowing System: macOS
![Screen Shot 2021-01-16 at 10 59 42 AM](https://user-images.githubusercontent.com/18336885/104816552-ff784280-57e9-11eb-8419-d89381df33bc.png)

**Comment by droidmonkey** [comment]
set the permission then restart the mac. This is an issue with Mac.

**Comment by jrobert-gh** [comment]
While that works around the issue, it isn't necessary with v2.6.2 on the same OS. Something in v2.6.3 is provoking it.

**Comment by droidmonkey** [comment]
No there isn't

**Comment by jrobert-gh** [comment]
Then how do you explain why the issue doesn't occur with 2.6.2 but it does with 2.6.3?

**Comment by droidmonkey** [comment]
Are you swapping between 2.6.2 and 2.6.3? If not, has the checkbox unchecked after s restart? The accessibility settings are 100% controlled by macOS. We don't set or unset accessibility settings.

**Comment by jrobert-gh** [comment]
No, not swapping. And I believe you about not changing the OS' accessibility settings.
All I can confirm is that with the setting set by me in SysPrefs with 2.6.3 not running, that running 2.6.3 and trying to Auto-type pops those dialogs, and I find the setting cleared.

When I then replaced (over-wrote) 2.6.3 with 2.6.2 and repeated the same trial, the setting now remains set, with no complaints about Auto-type. I did not upgrad to 2.6.3 again once I had downgraded.
